[CodeComplete] Include expr keywords in argument completion

hamishknight opened the pull request:
When completing for an argument expression, make sure to include keywords that can be used in an expression position, such as `try`, `await` and `super`.

rdar://77869845

hamishknight commented:
@swift-ci please test

rintaro commented on lib/IDE/CodeCompletion.cpp:
```diff
@@ -5952,7 +5953,8 @@ static void addAnyTypeKeyword(CodeCompletionResultSink &Sink, Type T) {
 void CodeCompletionCallbacksImpl::addKeywords(CodeCompletionResultSink &Sink,
-                                              bool MaybeFuncBody) {
+                                              bool MaybeFuncBody,
+                                              bool ForCallArgExpr) {
```
I don't think it's worth to call this function from the main `switch`. It can directly call
```cpp
      addSuperKeyword(CompletionContext.getResultSink());
      addExprKeywords(CompletionContext.getResultSink());
```
from the each `case` branch. What do you think?

hamishknight replied:
Yeah, that's fair enough

hamishknight committed:
[CodeComplete] Include expr keywords in argument completion

When completing for an argument expression, make
sure to include keywords that can be used in an
expression position, such as `try`, `await` and
`super`.

rdar://77869845

hamishknight commented:
@swift-ci please test

rintaro reviewed (approved):
Thank you!
